cpg_pipes/targets.py

```diff
@@ -0,0 +1,639 @@
+            + (f', pedigree={self.pedigree}' if self.pedigree else '')
+            + f')'
+        )
+
+    def __str__(self):
+        ai_tag = ''
+        if self.alignment_input:
+            if isinstance(self.alignment_input, CramPath):
+                if self.alignment_input.is_bam:
+                    ai_tag = f'|SEQ=CRAM'
+                else:
+                    ai_tag = f'|SEQ=BAM'
+            else:
+                ai_tag = f'|SEQ={len(self.alignment_input)}FQS'
+
+        ext_id = f'|{self._external_id}' if self._external_id else ''
+        return f'Sample({self.dataset.name}/{self.id}{ext_id}{ai_tag})'
+
+    @property
+    def participant_id(self) -> str:
+        """
+        Get ID of participant corresponding to this sample,
+        or substitute it with external ID.
+        """
+        return self._participant_id or self.external_id
+
+    @participant_id.setter
+    def participant_id(self, val: str):
+        """
+        Set participant ID.
+        """
+        self._participant_id = val
+
+    @property
+    def external_id(self) -> str:
+        """
+        Get external sample ID, or substitute it with the internal ID.
+        """
+        return self._external_id or self.id
+
+    def get_ped_dict(self, use_participant_id: bool = False) -> dict[str, str]:
+        """
+        Returns a dictionary of pedigree fields for this sample, corresponding
+        a PED file entry.
+        """
+        if self.pedigree:
+            return self.pedigree.get_ped_dict(use_participant_id)
+        return {
+            'Family.ID': self.participant_id if use_participant_id else self.id,
+            'Individual.ID': self.participant_id if use_participant_id else self.id,
+            'Father.ID': '0',
+            'Mother.ID': '0',
+            'Sex': '0',
+            'Phenotype': '0',
+        }
+
+    def get_cram_path(self) -> CramPath:
+        """
+        Path to a CRAM file. Not checking its existence here.
+        """
+        return CramPath(self.dataset.get_bucket() / 'cram' / f'{self.id}.cram')
+
+    def get_gvcf_path(self) -> GvcfPath:
+        """
+        Path to a GVCF file. Not checking its existence here.
+        """
+        return GvcfPath(self.dataset.get_bucket() / 'gvcf' / f'{self.id}.g.vcf.gz')
+
+    @property
+    def target_id(self) -> str:
+        """Unique target ID"""
+        return self.id
+
+    def get_samples(self, only_active: bool = True) -> list['Sample']:
+        """
+        Implementing the abstract method.
+        """
+        if only_active and not self.active:
+            return []
+        return [self]
+
+    def get_job_attrs(self) -> dict[str, str]:
+        """
+        Attributes for Hail Batch job.
+        """
+        return {'dataset': self.dataset.name, 'sample': self.id}
+
+    def get_job_prefix(self) -> str:
+        """
+        Prefix job names.
+        """
+        return f'{self.dataset.name}/{self.id}: '
+
+
+@dataclass
+class PedigreeInfo:
```

Comment:
Should we use the `peddy` classes here instead?

Reply:
Thought about it a bit. We indeed don't need PedigreeInfo at all. Created an issue:
https://github.com/populationgenomics/production-pipelines/issues/39